File: code/chapter3/ridge_boston.R
source('linear_regression_ridge.R')

library(MASS) # load Boston data from this package

# let's try lmabda = 0.5
ridge = LR_Ridge$new(0.5)
ridge$fit(data.matrix(Boston[,-ncol(Boston)]),Boston$medv)
print(ridge$coef)
# let's make prediction on the same data
pred=ridge$predict(data.matrix(Boston[,-ncol(Boston)]))
print(pred[1:5])

File: code/chapter4/ridge_boston.py
from sklearn.datasets import load_boston
from linear_regression_ridge import LR_Ridge
boston = load_boston()
X, y = boston.data, boston.target
# first, let's run our own linear regression
ridge = LR_Ridge(0.5)
ridge.fit(X, y)
print(ridge.coef)
print(ridge.predict(X)[:5])
